### PR TITLE
237 feat recommendations

### PR DESCRIPTION
**Pull request by Thenlie**
Added functions to get movie and tv show recommendations. These pull data from the MovieDB API and convert them to the showData type.

The types have also been updated slightly. We need to keep in mind that almost all fields are optional in the movieDB response as getting back null or undefined results could result in crashes without proper error handling.

- #237

### CODE REVIEW

File: src/types/tmdb.ts [hunk 1]
```diff
@@ -318,3 +318,10 @@ export interface ShowData {
         }
     ];
 }
+
+export interface RecommendationData {
```
Comment (maxh1231) [inline src/types/tmdb.ts]: Any reason not to use `interface MovieData` here? Or is this for readability/organization?

File: src/types/tmdb.ts [hunk 2]
```diff
@@ -319,9 +323,22 @@ export interface ShowData {
...
     page?: number;
     results?: MovieData[];
     total_pages?: number;
     total_results?: number;
 }
+
+/**
+ * Returned by MovieDB tv recommendation query
+ */
+export interface TvRecommendationData {
```
Comment (maxh1231) [inline src/types/tmdb.ts:339]: Same thing here for `TvShowData`.

**Review by maxh1231 — APPROVED** [review]
2 questions/nits. 

In this scenario, would I approve the PR with comments or request changes?

File: src/types/tmdb.ts (hunk 2 above)
Comment (Thenlie) [reply src/types/tmdb.ts:339]: TvShowData is contained in this response. We could use `getTvByName` though

File: src/types/tmdb.ts (hunk 1 above)
Comment (Thenlie) [reply src/types/tmdb.ts]: Updated to using ResultsData

File: src/types/tmdb.ts (hunk 2 above)
Comment (Thenlie) [reply src/types/tmdb.ts]: Updated

**Pull request merged**